From f352da2d607fffd2629ba408d1c9fb0da8c49c60 Mon Sep 17 00:00:00 2001
From: Kaylee Lubick <kjlubick@google.com>
Date: Fri, 24 Oct 2025 14:25:24 -0700
Subject: [PATCH] Reland "Make SkPath immutable on GN build"

This reverts commit a01eb11751829966ffe9925388e47f85ce585870.

Reason for revert: CanvasKit updated, PathKit deleted, WasmGM tests disabled

Original change's description:
> Revert "Make SkPath immutable on GN build"
>
> This reverts commit 832f6e513d41ec0b268dc1596169312b37f7a182.
>
> Reason for revert: Breaking Ubuntu builds
>
> Original change's description:
> > Make SkPath immutable on GN build
> >
> > Our clients don't use our GN files directly, so we can set the
> > define here and it should not bleed into those builds.
> >
> > Change-Id: Ia991a3075d2f551d8b264b1629548e21f0cbe303
> > Reviewed-on: https://skia-review.googlesource.com/c/skia/+/1074076
> > Auto-Submit: Kaylee Lubick <kjlubick@google.com>
> > Reviewed-by: Eric Boren <borenet@google.com>
> > Commit-Queue: Eric Boren <borenet@google.com>
>
> No-Presubmit: true
> No-Tree-Checks: true
> No-Try: true
> Change-Id: If5b99e4b73506bc5f31a189f0a9c3462486642bb
> Reviewed-on: https://skia-review.googlesource.com/c/skia/+/1078196
> Bot-Commit: Rubber Stamper <rubber-stamper@appspot.gserviceaccount.com>
> Commit-Queue: Thomas Smith <thomsmit@google.com>

Change-Id: I6caa8371ca86eeb51d8406bad9586f341ffac1e0
Reviewed-on: https://skia-review.googlesource.com/c/skia/+/1084436
Bot-Commit: Rubber Stamper <rubber-stamper@appspot.gserviceaccount.com>
Commit-Queue: Kaylee Lubick <kjlubick@google.com>
Signed-off-by: Hin-Tak Leung <htl10@users.sourceforge.net>
---
 BUILD.gn       | 3 +++
 gn/gn_to_bp.py | 7 ++++++-
 2 files changed, 9 insertions(+), 1 deletion(-)

diff --git a/BUILD.gn b/BUILD.gn
index 0fd3c5f93d..7645bf3dca 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -23,6 +23,9 @@ config("skia_public") {
   defines = [
     "SK_CODEC_DECODES_BMP",
     "SK_CODEC_DECODES_WBMP",
+
+    # TODO(kjlubick,fmalita) Remove when clients updated and APIs deleted.
+    "SK_HIDE_PATH_EDIT_METHODS",
   ]
   cflags_objcc = []
   if (is_component_build) {
diff --git a/gn/gn_to_bp.py b/gn/gn_to_bp.py
index 24195d2eea..2bbe758676 100755
--- a/gn/gn_to_bp.py
+++ b/gn/gn_to_bp.py
@@ -771,7 +771,12 @@ here = os.path.dirname(__file__)
 defs = gn_to_bp_utils.GetArchSources(os.path.join(here, 'opts.gni'))
 
 def get_defines(json):
-  return {str(d) for d in json['targets']['//:skia']['defines']}
+  defines = {str(d) for d in json['targets']['//:skia']['defines']}
+  # TODO(kjlubick, fmalita) Add this back in to enforce Android doesn't
+  # use these methods anymore.
+  defines.remove("SK_HIDE_PATH_EDIT_METHODS")
+  return defines
+
 android_defines      = get_defines(js)
 linux_defines        = get_defines(js_linux)
 mac_defines          = get_defines(js_mac)
-- 
2.52.0

